mmdet3d/models/dense_heads/groupfree3d_head.py

```diff
@@ -0,0 +1,1000 @@
+        dir_res_loss (dict): Config of direction residual regression loss.
+        size_class_loss (dict): Config of size classification loss.
+        size_res_loss (dict): Config of size residual regression loss.
+        size_reg_loss (dict): Config of class-agnostic size regression loss.
+        semantic_loss (dict): Config of point-wise semantic segmentation loss.
+    """
+
+    def __init__(self,
+                 num_classes,
+                 in_channels,
+                 bbox_coder,
+                 num_decoder_layers,
+                 transformerlayers,
+                 train_cfg=None,
+                 test_cfg=None,
+                 num_proposal=128,
+                 pred_layer_cfg=None,
+                 size_cls_agnostic=True,
+                 gt_per_seed=3,
+                 sampling_objectness_loss=None,
+                 objectness_loss=None,
+                 center_loss=None,
+                 dir_class_loss=None,
+                 dir_res_loss=None,
+                 size_class_loss=None,
+                 size_res_loss=None,
+                 size_reg_loss=None,
+                 semantic_loss=None):
+        super(GroupFree3DHead, self).__init__()
+        self.num_classes = num_classes
+        self.train_cfg = train_cfg
+        self.test_cfg = test_cfg
+        self.num_proposal = num_proposal
+        self.in_channels = in_channels
+        self.num_decoder_layers = num_decoder_layers
+        self.size_cls_agnostic = size_cls_agnostic
+        self.gt_per_seed = gt_per_seed
+
+        # Transformer decoder layers
+        if isinstance(transformerlayers, ConfigDict):
+            transformerlayers = [
+                copy.deepcopy(transformerlayers)
+                for _ in range(num_decoder_layers)
+            ]
+        else:
+            assert isinstance(transformerlayers, list) and \
+                   len(transformerlayers) == num_decoder_layers
+        self.decoder_layers = nn.ModuleList()
+        for i in range(self.num_decoder_layers):
+            self.decoder_layers.append(
+                build_transformer_layer(transformerlayers[i]))
+        self.embed_dims = self.decoder_layers[0].embed_dims
+
+        # bbox_coder
+        self.bbox_coder = build_bbox_coder(bbox_coder)
+        self.num_sizes = self.bbox_coder.num_sizes
+        self.num_dir_bins = self.bbox_coder.num_dir_bins
+
+        # Initial object candidate sampling
+        self.gsample_module = GeneralSamplingModule()
+        self.fps_module = Points_Sampler([self.num_proposal])
+        self.points_obj_cls = PointsObjClsModule(self.in_channels)
+
+        self.fp16_enabled = False
+
+        # initial candidate prediction
+        self.conv_pred = BaseConvBboxHead(
+            **pred_layer_cfg,
+            num_cls_out_channels=self._get_cls_out_channels(),
+            num_reg_out_channels=self._get_reg_out_channels())
+
+        # query proj and key proj
+        self.decoder_query_proj = nn.Conv1d(
```

Comment:
Shall we use mmcv ConvModule here as well? Asking for the opinion from @ZwwWayne